azurelinuxagent/common/exception.py

```diff
@@ -79,6 +79,15 @@ def __init__(self, msg=None, inner=None, code=-1):
         super(ExtensionOperationError, self).__init__(msg, inner, code)
 
 
+class ExtensionConfigurationError(ExtensionError):
+    """
+    When failed to configure an extension
+    """
```

Comment:
What do we mean by "failing to configure an extension"?
The current usage of the extension is for reporting bad configuration files... are you thinking of other uses for this exception?

Reply:
I was trying to ask what the intent of this exception is... that'll help people consuming it. 

"When failed... configuring the extension requirements for capabilities or cgroups" ... you can omit capabilities for the moment. As for cgroups, I see many places we throw CGroupsException when setting up cgroups. When should ExtensionConfigurationError be used instead?

Reply:
Made this exception more specific now `ExtensionHandlerConfigurationError`.